phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java

```diff
@@ -3048,7 +3048,13 @@ public boolean isViewReferenced() {
              * the counter as NULL_COUNTER for extra safety.
              */
             EncodedCQCounter cqCounterToBe = tableType == PTableType.VIEW ? NULL_COUNTER : cqCounter;
-            PTable table = new PTableImpl.Builder()
+            PTable table;
+            //better to use the table sent back from the server so we get an accurate DDL
+            // timestamp, which is server-generated.
+            if (result.getTable() != null ) {
```

Comment:
This seems risky since we are relying on the fact that the PTable returned from the server has all the necessary attributes set as the PTable we create on the client-side. There are some that we set explicitly inside `MetaDataClient` which depend on the parent so I'm not sure we still have those set as expected.

Instead, to be safe we can maybe `getDDLTimestamp()` from this returned PTable and set that in the builder. Better yet, we could just send the DDL timestamp in the server response rather than the entire PTable. We'd also have to add a setter and getter to the PTable builder for this new attribute.

Reply:
It seems strange that System.Catalog is the source of truth, we've just been there on the server, and yet the client copy controls. Will look closer at MetadataClient as you suggest to understand why. 

Are we missing tests? Because assuming the client-side copy _should_ control, my replacing it with the server-side copy should have broken tests somewhere, and it doesn't appear to.

Reply:
I'm not sure that would be covered by a test since we'd only see it if the test were to actually look at the cached PTable rather than say doing a `PhoenixRuntime.getTableNoCache()` right? Since we're changing what is cached, we should add some tests around this now to make sure the PTable has all expected fields set.

I see some tests that should cover this code path inside AlterTableWithViewsIT (like `testAlterPropertiesOfParentTable()`) and also some in ViewTTLIT, but I'm not sure to what extent they would capture this. 

I'm mostly worried about things like setting certain properties based on the parent PTable or overriding null properties with the default values, etc. (see [this](https://github.com/apache/phoenix/blob/ac0538b7f5456dd4f91b948c1a51c0042e115955/phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java#L3093-L3104) and [this](https://github.com/apache/phoenix/blob/ac0538b7f5456dd4f91b948c1a51c0042e115955/phoenix-core/src/main/java/org/apache/phoenix/schema/MetaDataClient.java#L3071-L3074)). That's where I'm not sure that the PTable returned from the server will be identical to the one that's created by the client today, since I don't see any code in MetaDataEndpointImpl that does this.